api/maven-api-model/src/main/mdo/maven.mdo

```diff
@@ -775,28 +775,30 @@
           <type>String</type>
         </field>
         <field>
-          <!-- TODO: replaced by <Source> with "main" scope and "resources" language -->
           <name>resources</name>
           <version>3.0.0+</version>
           <description>
             This element describes all the classpath resources such as properties
             files associated with a project. These resources are often included in the final
             package.
             The default value is {@code src/main/resources}.
+
+            @deprecated Replaced by {@code &lt;Source&gt;} with {@code main} scope and {@code resources} language.
           </description>
           <association>
             <type>Resource</type>
             <multiplicity>*</multiplicity>
           </association>
         </field>
```

Comment:
All deprecated elements should be added
```
          <annotations>
            <annotation>@Deprecated</annotation>
          </annotations>
```
This will add the usual `@Deprecated` annotation to the field and getter.

Reply:
Should we also replace `<version>3.0.0+</version>` by `<version>4.0.0</version>`? I see that other deprecated elements such as `module` have the latter.

Reply:
No the version here refers to the `modelVersion`, not Maven's version, so we don't want to change them.  New elements are introduced in model 4.1.0 which is fine.